File: RedSheet.DbEntities/Models/GlobalSetting.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("GlobalSettings", Schema = "dbo")]
	[Serializable]
	public partial class GlobalSetting  
	{
 
	[Required]
	public bool AutoTranslation { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ConfigurationId { get; set; } 
 
	[MaxLength(10)]
	public string LockDuration { get; set; } 

	public Nullable<int> PasswordPolicy { get; set; } 
 
	[Required]
	public bool RecordLock { get; set; } 
 
	[Required]
	public bool RequestLogging { get; set; } 
 
	[Required]
	public bool SocialAuth { get; set; } 
 
	[Required]
	public bool TwoFactorAuthentication { get; set; } 
 
	[RelationshipTableAttribue("ApplicationTimeZones","dbo")]
	//Relationships
	public int ApplicationTimeZoneId { get; set; } 
 
	[ForeignKey("ApplicationTimeZoneId")]
	public ApplicationTimeZone ApplicationTimeZone { get; set; } 
 
	[RelationshipTableAttribue("Languages","dbo")]
	//Relationships
	public int LanguageId { get; set; } 
 
	[ForeignKey("LanguageId")]
	public Language Language { get; set; } 

	public GlobalSetting()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/global-setting.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ApplicationTimeZone, Language,  } from './'
export class GlobalSetting {
    constructor(globalSetting?: GlobalSetting )  {
        let properties = [ "autoTranslation", "configurationId", "lockDuration", "passwordPolicy", "recordLock", "requestLogging", "socialAuth", "twoFactorAuthentication", "applicationTimeZoneId", "languageId",];
        for (let property of properties)
            if (globalSetting && globalSetting[property])
                this[property] = globalSetting[property];
    }
 
	autoTranslation : boolean = false ;
 
	configurationId : number =   0 ;
 
    @maxLength(10)
	lockDuration : string =   undefined;
 
	passwordPolicy : number =   undefined;
 
	recordLock : boolean = false ;
 
	requestLogging : boolean = false ;
 
	socialAuth : boolean = false ;
 
	twoFactorAuthentication : boolean = false ;
 
    @range(0,2147483647)
	applicationTimeZoneId : number =   undefined;
	applicationTimeZone : ApplicationTimeZone  ;
 
    @range(0,2147483647)
	languageId : number =   undefined;
	language : Language  ;


}
